apps/www/registry/default/ui/calendar.tsx

```diff
@@ -20,42 +19,83 @@ function Calendar({
-        day: cn(
-          buttonVariants({ variant: "ghost" }),
-          "h-9 w-9 p-0 font-normal aria-selected:opacity-100"
+        button_next: cn(
+          buttonVariants({
+            variant: "outline",
+            className:
+              "h-7 w-7 bg-transparent p-0 opacity-50 hover:opacity-100",
+          })
         ),
-        day_range_end: "day-range-end",
-        day_selected:
-          "bg-primary text-primary-foreground hover:bg-primary hover:text-primary-foreground focus:bg-primary focus:text-primary-foreground",
-        day_today: "bg-accent text-accent-foreground",
-        day_outside:
-          "day-outside text-muted-foreground opacity-50 aria-selected:bg-accent/50 aria-selected:text-muted-foreground aria-selected:opacity-30",
-        day_disabled: "text-muted-foreground opacity-50",
-        day_range_middle:
-          "aria-selected:bg-accent aria-selected:text-accent-foreground",
-        day_hidden: "invisible",
+        month_caption: "flex justify-center items-center h-7",
+        caption_label: "text-sm font-medium",
+        month_grid: "border-collapse space-y-1",
+        weekdays: "flex",
+        weekday: "text-muted-foreground w-9 font-normal text-xs",
+        weeks: "",
+        week: "flex mt-2",
+        day: "p-0",
+        outside: "bg-accent/40",
+        range_middle: "bg-accent last:rounded-e-md first:rounded-s-md",
+        range_start: "bg-accent rounded-s-md",
+        range_end: "bg-accent rounded-e-md",
         ...classNames,
       }}
       components={{
-        IconLeft: ({ ...props }) => <ChevronLeft className="h-4 w-4" />,
-        IconRight: ({ ...props }) => <ChevronRight className="h-4 w-4" />,
+        DayButton({ day, modifiers, className, ...buttonProps }) {
+          return (
+            <Button
+              variant={"ghost"}
+              className={cn(
+                className,
+                "h-9 w-9 p-0 font-normal",
+                modifiers?.today && "bg-accent text-accent-foreground",
+                modifiers?.selected &&
+                  "bg-primary text-primary-foreground hover:bg-primary hover:text-primary-foreground focus:bg-primary focus:text-primary-foreground",
+                modifiers?.outside &&
+                  "text-muted-foreground opacity-50 pointer-events-none",
+                modifiers?.disabled && "opacity-50 text-muted-foreground",
+                modifiers?.hidden && "invisible",
+                modifiers.range_middle &&
+                  "bg-accent text-accent-foreground hover:bg-accent hover:text-accent-foreground rounded-none last:rounded-e-md first:rounded-s-md",
+                modifiers.outside &&
+                  modifiers.selected &&
+                  "bg-accent/40 text-muted-foreground"
```

Comment:
If this is going to be pursued, the code could be made a bit more readable with the `ClassDictionary` from `clsx`. Instead of having these `&&` operators in every row, an object could be used like this:
```suggestion
                "bg-accent text-accent-foreground": modifiers?.today,
                "bg-primary text-primary-foreground hover:bg-primary hover:text-primary-foreground focus:bg-primary focus:text-primary-foreground": modifiers?.selected,
                "text-muted-foreground opacity-50 pointer-events-none": modifiers?.outside,
                "opacity-50 text-muted-foreground": modifiers?.disabled,
                invisible: modifiers?.hidden,
                "bg-accent text-accent-foreground hover:bg-accent hover:text-accent-foreground rounded-none last:rounded-e-md first:rounded-s-md": modifiers?.range_middle,
                "bg-accent/40 text-muted-foreground": modifiers?.outside && modifiers?.selected
```

Also, I noted that some `modifiers` values don't have the `?`. Don't know if it'd be a possible issue...

Reply:
I came across this PR while updating day picker in my project.

An object like what @McTom234 suggested is definitely cleaner but the previous code (e.g. `cell: ...` before it changed to `day`) was targeting `aria-selected` to add styling instead, maybe it would be better to keep that approach instead of importing `Button`? Anyways that's what I'll be doing.

Also I think lines like `"bg-primary text-primary-foreground hover:bg-primary hover:text-primary-foreground focus:bg-primary focus:text-primary-foreground": modifiers?.selected,` can be cleaned up since the default/hover/focus styles are the same, unless they're overriding styles set somewhere else.

Reply:
I agree on those lines, although the others are cleaner. I tried to upgrade to v9 myself, but didn't have the time to get a working solution done, so I stayed at v8 for now. But during the process, I felt like importing `Button` wasn't a big gain.

Anyways, if you have a working solution example, I'd appreciate a little insight :innocent:

Reply:
@McTom234 for me, following the [react-day-picker upgrading doc](https://daypicker.dev/upgrading) fixed most of the regressions. Seems like v9's HTML structure is different though so updating the nav buttons positioning isn't as straightforward. I'm working on that now but I'm using `nav: "flex items-start",` to have the nav float up to the top, `absolute left-2` and `absolute right-2` on `button_previous` and `button_next` respectively, and `months: "flex flex-col sm:flex-row", month: "m-4 mb-0 last-of-type:ml-0",` to adjust/fix some other layout regressions. Our calendar component is pretty heavily customized so I'm afraid that sharing it will be more noisy than helpful.

Reply:
Okay, thanks for your tips. I followed the same guide and had the issues with the nav buttons as well. That's where time-benefit wasn't given to proceed...